### PR TITLE
Prevent incorrect change in routeTable CR spec after adding route addition having vpcEndpointID

### PR DESCRIPTION
**Pull request by nnbu**
Fixes [#1935](https://github.com/aws-controllers-k8s/community/issues/1935)

Description of changes:
This fix accomodates an aws api bug so as to prevent routeTable CR spec from changing.
Details explained in the issue description.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl [hunk 1]
```diff
@@ -9,3 +9,15 @@
 		ko.Spec.Tags = r.ko.Spec.Tags
 	}
     
+	// Even if route is created with arguments as VPCEndpointID,
+	// when aws api is called to describe the route (inside skdFind), it
+	// returns VPCEndpointID as GatewayID. Due to this bug, spec section for
+	// routes is populated incorrectly in above auto-gen code.
+	// To solve this, if 'GatewayID' has prefix 'vpce-', then the entry is
+	// moved from 'GatewayID' to 'VPCEndpointID'.
```
Comment (a-hilaly) [inline templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: /cc @LikithaVemulapalli do you know how is this happening from the ec2 side? i'd prefer having EC2 team fix this bug than ec2-controller implementing a workaround (I don't think we'll be notified if/when they change this behaviour, which can cause more confusion for the users).
/hold

Comment (nnbu) [reply templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: @LikithaVemulapalli FYI
https://github.com/aws-controllers-k8s/community/issues/1935#issuecomment-1791358886 provides the details about the problem with aws api.

@a-hilaly Currently, routeTable CR is unusable when it uses route with `VPCEndpointID`. https://github.com/aws-controllers-k8s/community/issues/1935 mentions the issue with `update` of the CR only. But I realized later that this issue exists even in `creation` of the CR.

Comment (nnbu) [reply templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: Any update on this @LikithaVemulapalli ?

Comment (LikithaVemulapalli) [reply templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: > /cc @LikithaVemulapalli do you know how is this happening from the ec2 side? i'd prefer having EC2 team fix this bug than ec2-controller implementing a workaround (I don't think we'll be notified if/when they change this behaviour, which can cause more confusion for the users). /hold

Hello @a-hilaly @nnbu apologies for late response I was on vacation and started working a week ago. I'm not sure how this is happening from the ec2 side and I haven't worked on the route table create/update code as well, I agree with Amine and would prefer EC2 team fix this! Thank you.

Comment (nnbu) [reply templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: Hi @LikithaVemulapalli 
Is there any timeline for getting this fixed by EC2 team?

@a-hilaly 
If not by this PR, how can we address the issue until the fix comes from EC2 team? Currently, routeTable CR is unusable when it has `VPCEndpointID`.

Comment (a-hilaly) [reply templates/hooks/route_table/sdk_read_many_post_set_output.go.tpl:17]: Alright @nnbu - just ok-to-test this PR, i'll do one more round of reviews and merge ASAP. Thank you for all your contributions :)

**Review by a-hilaly — APPROVED** [review]
/lgtm

**Review by a-hilaly — APPROVED** [review]
👍 
/lgtm
/unhold

**Pull request merged**